### PR TITLE
Switch from error-chain to anyhow

### PR DESCRIPTION
**Pull request by bgilbert**
error-chain is no longer actively maintained.  Switch to a more modern library.  This does lose us backtrace support until `std::backtrace` is stabilized.

Switch to returning a `Result` directly from `main()`, since anyhow has reasonable default error rendering.

### CODE REVIEW

**Comment by bgilbert** [comment]
Ready for review.

File: src/blockdev.rs [hunk 1]
```diff
@@ -322,18 +321,23 @@ impl Partition {
         // We multiply by 512 here: the kernel values are always in 512 blocks, regardless of the
         // actual sector size of the block device. We keep the values as bytes to make things
         // easier.
-        let start_offset: u64 = start.checked_mul(512).ok_or("start offset mult overflow")?;
+        let start_offset: u64 = start
+            .checked_mul(512)
+            .ok_or_else(|| anyhow!("start offset mult overflow"))?;
```
Comment (jlebon) [inline src/blockdev.rs:326]: Hmm, interesting. Does `ok_or("string")` not work anymore? I would think the implied `into()` would convert it fine.

**Review by jlebon — APPROVED** [review]
I only scanned this, but it looks sane to me!

File: src/blockdev.rs (hunk 1 above)
Comment (bgilbert) [reply src/blockdev.rs:326]: It doesn't.  `anyhow::Error` doesn't implement `From<&str>`.

Comment (jlebon) [reply src/blockdev.rs:326]: Ack.

**Pull request merged**